scripts/calculateGenreRecs.js

```diff
@@ -0,0 +1,101 @@
+const postgresDb = require('../postgresDb/index.js');
+const mongoDb = require('../mongoDb/index.js');
+
+const genreArr = [
+  'action',
+  'animation',
+  'comedy',
+  'documentary',
+  'drama',
+  'family',
+  'fantasy',
+  'international',
+  'horror',
+  'musical',
+  'mystery',
+  'romance',
+  'sci_fi',
+  'thriller',
+  'western',
+];
+
+const updateGenreRecs = (genre) => {
+  postgresDb.Movie.findAll({
+    // find top 1000 movies by view count
+    order: [['views', 'DESC']],
+    limit: 1000,
+  })
+    .then(result => (
+      // return [[movieId, score],...]
+      result.map((movie) => {
+        const m = movie.dataValues;
+        const prof = JSON.parse(m.profile);
+        return [m.id, m.views * prof[genre]];
+      })
+    ))
+    .then(scores => (
+      // sort scores in descending order
+      scores.sort((a, b) => {
+        if (a[1] > b[1]) {
+          return -1;
+        } else if (a[1] < b[1]) {
+          return 1;
+        }
+        return 0;
+      })
+    ))
+    .then((sortedDesc) => {
+      // return array of top 20 movieIds
+      const mostPopularIds = [];
+      for (let i = 0; i < 20; i += 1) {
+        mostPopularIds.push(sortedDesc[i][0]);
+      }
+      // look up movies by ids
+      return postgresDb.Movie.findAll({
```

Comment:
Just to clarify, the two function bodies above this are synchronous, so both of those can be refactored into one function block that end with returning this query.

Reply:
done